lib/graphql/enum_type.rb

```diff
@@ -67,6 +67,7 @@ def validate_non_null_input(value_name)
     # @param value_name [String] the string representation of this enum value
     # @return [Object] the underlying value for this enum value
     def coerce_non_null_input(value_name)
+      return nil unless @values_by_name.keys.include?(value_name)
       @values_by_name.fetch(value_name).value
```

Comment:
How about using `@values_by_name[value_name]` instead of explicitly checking if the key is included.  That will be more efficient than creating a keys array then iterating over it to see if the value name is included.